### DISCUSSION TITLE
Quantization-aware training

### DISCUSSION THREAD

**Discussion by jeremy-u** [keras-team/keras#19704] [Q&A] [answered]
I'm looking for some guidance on how to do quantization-aware training in keras 3.  I have used the TF model-optimization library in the past, but per the responses on [this issue](https://github.com/tensorflow/model-optimization/issues/1119) it does not look like there is a plan to support tfmot+keras3.  Is there a go-to 3rd party library?  Any suggestions?

**Comment by abhaskumarsinha** [comment]
Any idea? @james77777778 if you are free can you point out library for that?

**Reply by james77777778** [reply, marked as answer]
I think the best option for now is to use Keras2 for TFMOT.
I have considered developing a library similar to TFMOT to support Keras 3, but currently, I don't have the bandwidth for it.

**Reply by abhaskumarsinha** [reply]
@james77777778  
Okay. Thank you for the idea. 🙏
Got it.

**Reply by jeremy-u** [reply]
Thanks James! I'll stick with Keras 2 for now.
